subsys/fs/nvs/nvs.c

```diff
@@ -483,7 +516,7 @@ static int nvs_gc(struct nvs_fs *fs)
 
 	stop_addr = gc_addr - ate_size;
 
-	if (!nvs_ate_crc8_check(&close_ate)) {
+	if (nvs_close_ate_valid(fs, &close_ate)) {
```

Comment:
lack of check of case close_ate.offset > (fs->sector_size - ate_size)
need add service of close_ate.offset == (fs->sector_size - ate_size)

Reply:
IMO not needed, see comment above.